### PR TITLE
docs: add maintainer guide

### LINKED ISSUE #5238: Docs: Add a Maintainer Guide / Maintenance section to the website
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Surfacing a discussion with @bradzacher: we really like how the ESlint website includes a maintainer guide at https://eslint.org/docs/latest/maintainer-guide. It's good form to document these things.

Going back through recent issues...
* Release cadence & strategies
* What constitutes a breaking change
* What to do for new TS versions
* What kinds of bug fixes & feature requests generally will or won't be accepted
* Philosophy of when we'll allow fixes and/or suggestions (e.g. #5228)
* Edit: how to write an extension rule (e.g. the testing practice mentioned in #5183)

### Affected URL(s)

typescript-eslint.io/docs/maintenance, perhaps?

### PR DESCRIPTION
**Pull request by bradzacher**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #5238
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Adds maintainer/contributor guides for the website.

Very much a WIP - far from complete or finished.

- [x] Versioning and Releases (1st draft)
- [ ] Managing issues 
  - [ ] Guide for bug fixes
  - [ ] Guide for new rules
  - [ ] Guide for rule changes
- [ ] New TS release actions

### CODE REVIEW

File: docs/contributor/MANAGING_ISSUES.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+Before you being reviewing a simple bug report you should first ensure that there is a complete, isolated reproduction case provided by the reporter. This means that there should be a playground link which shows the issue being reported. Whilst a playground link is definitely preferred; it is not explicitly required if there is code and config in the issue that can be pasted into the playground to reproduce the issue. If you cannot reproduce the issue in the playground, remove the "triage" tag, add the "awaiting response" tag, and ask the user to create an isolated reproduction in the playground and comment back when they have one. For example:
+
+> Thanks for the report `@reporter`. Unfortunately I couldn't reproduce the issue using the code you provided. Could you please create an isolated reproduction in our playground (https://typescript-eslint.io/play/) and comment back when you've got one? We prefer an isolated reproduction so that we as volunteer maintainers can quickly reproduce the issue and more easily find the cause.
```
Comment (JoshuaKGoldberg) [inline docs/contributor/MANAGING_ISSUES.md:21]: Aside: one of the projects on my shortlist is to have a `.github/messages.json` file or similar that can power a browser extension to fill these in quickly. Maybe such a thing exists already, or maybe I'll have to write it.

Comment (Josh-Cena) [reply docs/contributor/MANAGING_ISSUES.md:21]: GitHub saved replies? (I have a deja vu that we talked about it...)

Comment (JoshuaKGoldberg) [reply docs/contributor/MANAGING_ISSUES.md:21]: Yes! Thank you haha, that's the thing. 

And they have good docs now! https://docs.github.com/en/get-started/writing-on-github/working-with-saved-replies/using-saved-replies

Comment (bradzacher) [reply docs/contributor/MANAGING_ISSUES.md:21]: I have a few saved replies myself yeah. I wonder if we can just add them to a wiki to allow us to quickly share them between one another?

Comment (JoshuaKGoldberg) [reply docs/contributor/MANAGING_ISSUES.md:21]: https://github.com/JoshuaKGoldberg/refined-saved-replies

I feel like a hacker again 😎 🥲

Comment (JoshuaKGoldberg) [reply docs/contributor/MANAGING_ISSUES.md:21]: https://github.com/typescript-eslint/typescript-eslint/issues/5547

**Comment by bradzacher** [comment]
superseded by 5874

**Pull request closed without merging**